x/programs/rust/sdk-macros/src/lib.rs

```diff
@@ -18,105 +19,89 @@ pub fn public(_: TokenStream, item: TokenStream) -> TokenStream {
     let input = parse_macro_input!(item as ItemFn);
 
     let vis_err = if !matches!(input.vis, Visibility::Public(_)) {
-        let err = Error::new(
+        Err(Error::new(
             input.sig.span(),
             "Functions with the `#[public]` attribute must have `pub` visibility.",
-        );
-
-        Some(err)
+        ))
     } else {
-        None
+        Ok(())
     };
 
-    let (input, user_specified_context_type, first_arg_err) = {
-        let mut context_type: Box<Type> = Box::new(parse_str(CONTEXT_TYPE).unwrap());
-        let mut input = input;
-
-        let first_arg_err = match input.sig.inputs.first_mut() {
-            Some(FnArg::Typed(PatType { ty, .. })) if is_context(ty) => {
-                let types = (context_type.as_mut(), ty.as_mut());
-
-                if let (Type::Path(context_type), Type::Path(ty)) = types {
-                    let args = [context_type, ty].map(|type_path| {
-                        type_path
-                            .path
-                            .segments
-                            .last_mut()
-                            .map(|segment| &mut segment.arguments)
-                    });
-
-                    if let [Some(context_type_args), Some(ty_args)] = args {
-                        *context_type_args = ty_args.clone();
-                    }
-                }
-
-                std::mem::swap(&mut context_type, ty);
-                None
+    let mut input = input;
+    let user_specified_context_type = match input.sig.inputs.first_mut() {
+        Some(FnArg::Typed(pat_type)) if is_context(&pat_type.ty) => {
+            // pass parameter if ignored with underscore
+            if matches!(*pat_type.pat, Pat::Wild(_)) {
+                pat_type.pat = Box::new(Pat::Ident(PatIdent {
+                    attrs: vec![],
+                    by_ref: None,
+                    mutability: None,
+                    ident: Ident::new("_ctx", Span::call_site()),
+                    subpat: None,
+                }))
```

Comment:
This is now needed because the parameters are all user-defined. Meaning that the context that used to be `param_0` as expected could also be a wildcard `_` and this parameter cannot be passed around.
Now the issue is that this is going to fail if any other parameter is named `_ctx` such as with this function signature:
```rust
#[public]
pub fn inc(_: Context, _ctx: Context, external: Program, address: Address)
```